src/codeSigning/packAndSign.ts

```diff
@@ -322,82 +198,29 @@ export const api = {
     return fs.writeFile(pathGetter.packageJson, JSON.stringify(pJson, null, 4));
   },
 
-  async doSign(args: SigningOpts, packageJson?: { name: string; version: string }): Promise<SigningResponse> {
-    const logger = await Logger.child('packAndSign');
-    const filepath = args.tarpath;
-
-    if (!filepath) {
-      throw new NamedError('MissingTarFile', 'A tgz file to sign was not specified.');
-    }
-
-    // create the signature file
-    const signature = await api.retrieveSignature(
-      createReadStream(filepath, { encoding: 'binary' }),
-      createReadStream(args.privatekeypath)
-    );
-    logger.debug('created the digital signature');
-    if (signature && signature.length > 0) {
-      // write the signature file to disk
-      const sigFilename = await api.writeSignatureFile(filepath, signature);
-
-      cliUx.log(`Artifact signed and saved in ${sigFilename}`);
-
-      let verified: boolean;
-      try {
-        // verify the signature with the public key url
-        verified = await api.verify(
-          createReadStream(filepath, { encoding: 'binary' }),
-          createReadStream(sigFilename),
-          args.publickeyurl
-        );
-      } catch (e) {
-        cliUx.error(e);
-        throw new NamedError(
-          'VerificationError',
-          'An error occurred trying to validate the signature. Check the public key url and try again.',
-          e
-        );
-      }
-      if (verified) {
-        cliUx.log(`Successfully verified signature with public key at: ${args.publickeyurl}`);
-        return {
-          tarPath: filepath,
-          filename: sigFilename,
-          verified,
-          name: packageJson?.name,
-          version: packageJson?.version,
-        };
-      } else {
-        // noinspection ExceptionCaughtLocallyJS
-        throw new NamedError('FailedToVerifySignature', 'Failed to verify signature with tar gz content');
-      }
-    } else {
-      // noinspection ExceptionCaughtLocallyJS
-      throw new NamedError('EmptySignature', 'The generated signature is empty. Verify the private key and try again');
+  async revertPackageJsonIfExists(): Promise<void> {
```

Comment:
Does it make more sense to put this in the `Package` class?

Reply:
Maybe, but it's 1) used in the file 2) similar to the other things that are in here like `copyPackageDotJson` and `writePackageJson` and using `pathGetter` from in here.  So all that would be a bigger change than just that one.

Plus we'd be constructing an entire `Package` just to copy a file.

There's a whole bunch of stuff on Package that seems redundant with things in here (it has readPackageJson, writePackageJson, etc.)

Thoughts?